fetcher/errors.go

```diff
@@ -55,7 +55,7 @@ func (f *Fetcher) RequestFailedError(
 	}
 
 	return &Error{
-		Err:       fmt.Errorf("%w: %s %s", ErrRequestFailed, message, err.Error()),
+		Err:       fmt.Errorf("%w: %s %s", ErrRequestFailed, message, err),
```

Comment:
did you mean to change this to: `fmt.Errorf("%w: %s %s", err, message, ErrRequestFailed)`? Mapping %w to original error, will preserve the error chain.

Reply:
exactly, will add test asap!